### PR TITLE
refactor!: rename renovate/stability-days check to renovate/minimum-release-age

### LINKED ISSUE #21442: Change the `renovate/stability-days` status check to `renovate/minimum-release-age`
### Describe the proposed change(s).

Creating an issue to track this idea.

### Original comments leading to this issue

> > Document that the minimumReleaseAge config option causes a renovate/stabilityDays option to be added. This would be very unintuitive to new users of the feature otherwise.
> 
> Is changing the `renovate/stability-days` status check to `renovate/minimum-release-age` also planned? I know that it would have to be done separately as it would be a breaking change

Planned but I don't think we have an issue yet. I would like to think if it's the right time to let users customize Renovate's status check strings too

_Originally posted by @rarkins in https://github.com/renovatebot/renovate/issues/21376#issuecomment-1504727770_

### Todos:

- [ ] #21376
- [ ] Change the `renovate/stability-days` status check to `renovate/minimum-release-age`
- [ ] _Optional:_ let users customize Renovate's status check strings

### PR DESCRIPTION
**Pull request by rarkins**
## Changes

Renames `renovate/stability-days` check to be `renovate/minimum-release-age`.

## Context

Closes #21442

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -2179,23 +2179,50 @@ Use the syntax `!/ /` like this:
 }
 ```
 
-### matchFiles
+### matchFileNames
```
Comment (viceice) [inline docs/usage/configuration-options.md:2182]: wrong merge?

Comment (rarkins) [reply docs/usage/configuration-options.md]: Try again now

File: lib/workers/repository/update/branch/index.ts [hunk 2]
```diff
@@ -333,7 +333,7 @@ export async function processBranch(
     ) {
       // Only set a stability status check if one or more of the updates contain
       // both a minimumReleaseAge setting and a releaseTimestamp
-      config.stabilityStatus = 'green';
+      config.ageStatus = 'green';
```
Comment (viceice) [inline lib/workers/repository/update/branch/index.ts:336]: was it a templateable field? so it's breaking there?

File: lib/workers/repository/update/branch/status-checks.ts [hunk 3]
```diff
@@ -54,24 +54,24 @@ async function setStatusCheck(
 }
 
 export interface StabilityConfig extends RenovateConfig {
```
Comment (viceice) [inline lib/workers/repository/update/branch/status-checks.ts:56]:
```suggestion
 export interface AgeConfig extends RenovateConfig {
```
rename too?

File: lib/workers/repository/update/branch/index.ts (hunk 2 above)
Comment (rarkins) [reply lib/workers/repository/update/branch/index.ts:336]: No, it's not

**Review by viceice — DISMISSED** [review]

File: docs/usage/configuration-options.md [hunk 4]
```diff
@@ -1602,7 +1602,7 @@ If you are more interested in including only certain package managers (e.g. `npm
```
[the remark this answers is not in the document]
Comment (rarkins) [reply docs/usage/configuration-options.md:1602]: Well, it's in a major release, however app users won't have a choice. We could set _both_ for a while, but that probably doesn't help things

[the remark this answers is not in the document]
Comment (HonkingGoose) [reply docs/usage/configuration-options.md:1602]: Having _both_ names is probably more confusing than a breaking rename to `renovate/minimum-release-age`. But in the short term having _both_ is probably needed for users to switch without breakage to the new check.

How about something like this?

1. Create the new check `renovate/minimum-release-age`
2. Create a notice about "Switch to `renovate/minimum-release-age` check" in:
    - a pinned Issue
    - a pinned Discussion
    - our docs announcement bar
3. After enough time remove the old `renovate/stability-days` check

Optional: use GitHub's brown-out strategy, they sometimes break a feature intentionally, and put a comment in the log, so people go looking why things changed. That way they nudge user into updating before the feature is fully removed by GitHub.

**Pull request closed without merging**